### PR TITLE
Remove uneeded role attributes

### LINKED ISSUE #1750: HTML validation issues
Caldera Forms v1.5.2.1 on WP v4.8 

Could you please fix html validation issues, I have attached a patch for review.  Here are the issues.

- "field" is not a valid value for the "role" attribute.
- "role" attribute with the value of "form" is not needed on a "form" element.
- When a label is hidden for a particular field, "labelledby" attribute is mislabeled. 

[caldera-forms.txt](https://github.com/CalderaWP/Caldera-Forms/files/1149285/caldera-forms.txt)

### PR DESCRIPTION
**Pull request by Shelob9**
Fixes #1750

### CODE REVIEW

File: classes/core.php [hunk 1]
```diff
@@ -4119,7 +4119,6 @@ static public function render_form( $atts, $entry_id = null, $shortcode = null )
 		$form_attributes = array(
 			'method'	=>	'POST',
 			'enctype'	=>	'multipart/form-data',
-			'role'		=>	'form',
```
Comment (Shelob9) [inline classes/core.php]: For #1750

**Comment by New0** [comment]
@Shelob9 Are we sure we want to remove the role attribute from form element ? 
It is not necessary but tools like JAWS use it. It makes the form to be interpreted as a region and some people are used to navigate with the R shortcut

**Comment by Shelob9** [comment]
I consulted with @bamadesigner who knows better than I do here. She says: `https://www.w3.org/TR/wai-aria-practices/examples/landmarks/form.html
"If a form element has an accessible name it is considered form landmark."`

Since we do not have a label for forms, I think we should not output a a role attribute, but introduce the filter so that Easy Pods and Easy Queries can set `role="search"`

**Comment by bamadesigner** [comment]
Amanda is correct. Per that Stack Overflow question, you should always follow the standards and not what 1 specific piece of software does. <form>s do not need `role="form".`

Also, don't forget the aria-label for the form so ARIA does know what the form is for and the form has an accessible name.

**Comment by Shelob9** [comment]
Ok, so I'm sold on removing role attribute. I actually think we can get the role=search in for Easy Pods/ Easy Queries search forms without a new filter. 

@bamadesigner Your point about aria-label I agree with, but out of scope of this PR. I created  a new issue #2093

**Comment by New0** [comment]
Ok me too @Shelob9 , @bamadesigner convinces me, we better follow the standards

**Comment by Shelob9** [comment]
I have updated the docs for the caldera_forms_render_form_attributes filter with an example of how to set the role attribute to search.

https://calderaforms.com/doc/caldera_forms_render_form_attributes/

**Pull request merged**